packages/graphql/src/schema/resolvers/field/get-cypher-result-variables.ts

```diff
@@ -24,14 +24,14 @@ export async function getCypherResultVariables(statement: string, executor: Exec
     const explainStatement = wrapInExplain(statement);
     const result = await executor.execute(explainStatement, {}, "READ");
     if (!result.summary.plan) throw new Error("Failed to get the plan of @cypher");
-    return result.summary.plan.identifiers.filter((v) => !variableIsLiteralValue(v));
+
+    const plan = result.summary.plan;
+    if (!plan.operatorType.match(/ProduceResults@.*/)) {
+        return [];
+    }
+    return [plan.arguments["Details"]];
```

Comment:
it may be worth extracting this to a dedicated function that parses the result of the `EXPLAIN`. It would also allow for the logic to grow better.